src/scene/frame-graph.js

```diff
@@ -24,7 +24,28 @@ class FrameGraph {
      */
     addRenderPass(renderPass) {
         Debug.assert(renderPass);
-        this.renderPasses.push(renderPass);
+
+        const passes = this.renderPasses;
+
+        const beforePasses = renderPass.beforePasses;
```

Comment:
Why not?
`passes = renderPass.beforePasses.filter(p => p.enabled)`

Reply:
this runs every frame, so performance is the reason
https://plainenglish.io/blog/are-for-loops-better-than-arrays-filter-or-foreach-methods